connectors/sources/s3.py

```diff
@@ -26,9 +26,11 @@
 DEFAULT_MAX_FILE_SIZE = 10485760
 DEFAULT_PAGE_SIZE = 100
 DEFAULT_CONTENT_EXTRACTION = True
-ENDPOINT_URL = os.getenv(
-    key="ENDPOINT_URL", default=None
-)  # For end to end tests set this environment variable with target host.
+
+if "AWS_ENDPOINT_URL" in os.environ:
+    AWS_ENDPOINT = f"{os.environ['AWS_ENDPOINT_URL']}:{os.environ['AWS_PORT']}"
+else:
+    AWS_ENDPOINT = None
```

Comment:
What does the `None` case mean?

Reply:
I guess when it's null it uses real AWS?

Reply:
yes. see https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session.client